app/controllers/application_controller.rb

```diff
@@ -44,17 +44,27 @@ def permit_params(*params)
   end
 
   def authorize!
-    authorization = AuthToken.decode(request.headers["Authorization"])
-
-    if authorization
-      user_id = authorization.first.with_indifferent_access[:user_id]
-      return unless user_id # requested is authorized if using generic JWT
-      return if User.find_by(id: user_id)&.admin
-    end
+    # requested is authorized if using generic JWT (current_user_id will be nil)
+    return if authorization && (current_user_id.nil? || current_user&.admin)
```

Comment:
I'm intrigued by this returning authorized if `current_user_id` is nil. To me that seems like a potential security loophole, in what circumstances would the JWT have a nil user id?

I do see the check from the previous version of the code, I'm just curious why this functionality exists.

Reply:
No idea. Just keeping things unchanged, assuming there was some good reason for it.

Reply:
Spencer was the one who added it, it might have been related to bridging the old auth mechanism to the new one that actually has users. If that is the case we can drop this scenario.

Reply:
it looks like an AuthToken can be created without a user here: https://github.com/CruGlobal/mobile-content-api/pull/469/files#diff-dfbe97f70beada492206e82fea7ed6fe79afdf33449579101b1d44effbe8108dR18-R24

this is the old authentication method that we can actually remove. [Here](https://jira.cru.org/browse/GT-1348) is a jira ticket I created to track removing the auth mechanism.

Reply:
For this PR let's remove the `current_user_id.nil?` condition, the user should only be authorized for admin resources if they are an actual user and are an admin.